crates/astria-sequencer/src/ibc/ics20_transfer.rs

```diff
@@ -430,4 +474,160 @@ mod test {
+        let storage = cnidarium::TempStorage::new()
+            .await
+            .expect("failed to create temp storage backing chain state");
+        let snapshot = storage.latest_snapshot();
+        let mut state_tx = StateDelta::new(snapshot.clone());
+
+        let bridge_address = Address::from([99; 20]);
+        let rollup_id = RollupId::from_unhashed_bytes(b"testchainid");
+        let denom: Denom = "nootasset".to_string().into();
+
+        state_tx.put_bridge_account_rollup_id(&bridge_address, &rollup_id);
+        state_tx
+            .put_bridge_account_asset_ids(&bridge_address, &[denom.id()])
+            .unwrap();
+
+        let packet = FungibleTokenPacketData {
+            denom: "nootasset".to_string(),
+            sender: String::new(),
+            amount: "100".to_string(),
+            receiver: hex::encode(bridge_address),
+            memo: "destinationaddress".to_string(),
+        };
+        let packet_bytes = serde_json::to_vec(&packet).expect("failed to serialize packet data");
+
+        execute_ics20_transfer(
+            &mut state_tx,
+            &packet_bytes,
+            &"source_port".to_string().parse().unwrap(),
+            &"source_channel".to_string().parse().unwrap(),
+            &"dest_port".to_string().parse().unwrap(),
+            &"dest_channel".to_string().parse().unwrap(),
+            false,
+        )
+        .await
+        .unwrap();
+
+        let denom: Denom = format!("dest_port/dest_channel/{}", "nootasset").into();
+        let balance = state_tx
+            .get_account_balance(bridge_address, denom.id())
+            .await
+            .unwrap();
+        assert_eq!(balance, 100);
+
+        let deposit = state_tx.get_block_deposits().await.unwrap();
+        assert_eq!(deposit.len(), 1);
+    }
+
+    #[tokio::test]
+    async fn execute_ics20_transfer_to_bridge_account_invalid() {
+        let storage = cnidarium::TempStorage::new()
+            .await
+            .expect("failed to create temp storage backing chain state");
+        let snapshot = storage.latest_snapshot();
+        let mut state_tx = StateDelta::new(snapshot.clone());
+
+        let bridge_address = Address::from([99; 20]);
+        let rollup_id = RollupId::from_unhashed_bytes(b"testchainid");
+        let denom: Denom = "nootasset".to_string().into();
+
+        state_tx.put_bridge_account_rollup_id(&bridge_address, &rollup_id);
+        state_tx
+            .put_bridge_account_asset_ids(&bridge_address, &[denom.id()])
+            .unwrap();
+
+        // use empty memo, which should fail
+        let packet = FungibleTokenPacketData {
+            denom: "nootasset".to_string(),
+            sender: String::new(),
+            amount: "100".to_string(),
+            receiver: hex::encode(bridge_address),
+            memo: String::new(),
+        };
+        let packet_bytes = serde_json::to_vec(&packet).expect("failed to serialize packet data");
```

Comment:
This line states that you expec tthe serialization to fail. Is that right?

Reply:
should be the opposite lol, will clarify